packages/engine/tests/simulations/experiment.rs

```diff
@@ -0,0 +1,277 @@
+use std::{
+    collections::HashMap,
+    fs::{self, File},
+    io::BufReader,
+    path::{Path, PathBuf},
+};
+
+use error::{bail, ensure, report, Result, ResultExt};
+use hash_engine::utils::OutputFormat;
+use regex::Regex;
+use serde::{de::DeserializeOwned, Deserialize};
+use serde_json::Value;
+
+pub type AgentStates = Value;
+pub type Globals = Value;
+pub type Analysis = Value;
+
+#[derive(Deserialize)]
+#[serde(rename_all = "kebab-case")]
+pub struct ExpectedOutput {
+    #[serde(default)]
+    pub json_state: HashMap<String, AgentStates>,
+    #[serde(default)]
+    pub globals: Option<Globals>,
+    #[serde(default)]
+    pub analysis_outputs: Option<Analysis>,
+}
+
+pub enum Experiment {
+    Simple { experiment: String },
+    SingleRun { steps: u64 },
+}
+
+impl Experiment {
+    pub fn run(&self, project: impl AsRef<Path>) -> Result<Vec<(AgentStates, Globals, Analysis)>> {
+        let output = std::env::var("OUT_DIR").wrap_err("$OUT_DIR is not set")?;
+        let manifest_path =
+            std::env::var("CARGO_MANIFEST_DIR").wrap_err("$CARGO_MANIFEST_DIR is not set")?;
+        let manifest_path = Path::new(&manifest_path);
+
+        let mut cmd = std::process::Command::new(manifest_path.join("target/debug/cli"));
+        cmd.env("RUST_LOG", "trace")
+            .arg("--project")
+            .arg(project.as_ref())
+            .arg("--output")
+            .arg(output)
+            .arg("--emit")
+            .arg(OutputFormat::Full.to_string())
+            .arg("--num-workers")
+            .arg("1");
+
+        match self {
+            Experiment::SingleRun { steps } => {
+                cmd.arg("single-run")
+                    .arg("--num-steps")
+                    .arg(steps.to_string());
+            }
+            Experiment::Simple { experiment } => {
+                cmd.arg("simple").arg("--experiment-name").arg(experiment);
+            }
+        }
+
+        let experiment = cmd.output().wrap_err("Could not run experiment command")?;
+        if !experiment.stdout.is_empty() {
+            println!("{}", String::from_utf8_lossy(&experiment.stdout));
+        }
+        if !experiment.stderr.is_empty() {
+            eprintln!("{}", String::from_utf8_lossy(&experiment.stderr));
+        }
+
+        ensure!(experiment.status.success(), "Could not run experiment");
+
+        let mut outputs = Regex::new(r#"Making new output directory: "(.*)""#)
+            .wrap_err("Could not compile regex")?
+            .captures_iter(&String::from_utf8_lossy(&experiment.stderr))
+            .map(|output_dir_capture| PathBuf::from(&output_dir_capture[1]))
+            .collect::<Vec<_>>();
+        outputs.sort_unstable();
```

Comment:
Yes, the outputs may be logged in any order so we have to sort them to match the `"expected_output"`. As soon as we split the CLI into binary+library we won't need this anymore

Reply:
It's sorting as if it's a string correct? What will happen with
"{my_path}/10"
"{my_path}/7"
"{my_path}/1"

That example might not be a good one, but folder sorting often struggles with double digit numbers because it groups by digit rather than by value. Unsure what will happen here

Reply:
Good point, this would be ordered uncorrectly

Reply:
Not a nice solution but as we plan to replace this section as soon the CLI got splitted up, I'm fine with it.

Reply:
I'd just parse the final component as an int and sort that. I don't think we intend to change the simulation run folders to a non-int so it's a fine assumption and less hacky than this haha.

Reply:
It's a bit more code but it's definitely the cleaner way of doing so. 
... done